src/main/scala/nl/knaw/dans/easy/pid/generator/PidManagerComponent.scala

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (C) 2015 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.pid.generator
+
+import java.sql.{ Connection, SQLException }
+
+import nl.knaw.dans.easy.pid._
+import org.joda.time.DateTime
+
+import scala.util.{ Failure, Try }
+
+trait PidManagerComponent {
+  this: DatabaseComponent with DatabaseAccessComponent =>
+
+  val pidManager: PidManager
+
+  class PidManager(formatters: Map[PidType, PidFormatter]) {
+
+    /**
+     * Generates the next PID of the specified type.
+     *
+     * @param pidType the type of PID to generate (DOI or URN)
+     * @return the PID
+     */
+    def generate(pidType: PidType)(implicit connection: Connection): Try[Pid] = {
```

Comment:
is anything logged into a log-file for initializing or generating?

Reply:
yes, that is done in `PidGeneratorApp`

I also added logging the errors there, as you requested without stacktrace for `PidNotInitialized`, `DuplicatePid` and `PidAlreadyInitialized`. The `DatabaseException` and all other exceptions that may occur, are still logged with a stacktrace though, since they represent an actual failure of the application.

Reply:
if you want more logging, let me know!